ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -404,17 +424,27 @@ private void checkPreconditions() {
     }
   }
 
-  private void waitForPreconditions() {
+  private void waitForPreconditions() throws AbortApplicationStartException {
     final List<Predicate<Exception>> predicates = ImmutableList.of(
         e -> !(e instanceof KsqlFailedPrecondition)
     );
-    RetryUtil.retryWithBackoff(
-        Integer.MAX_VALUE,
-        1000,
-        30000,
-        this::checkPreconditions,
-        predicates
-    );
+    try {
+      RetryUtil.retryWithBackoff(
+          Integer.MAX_VALUE,
+          1000,
+          30000,
+          this::checkPreconditions,
+          shuttingDown,
+          predicates
+      );
+    } catch (KsqlFailedPrecondition e) {
+      log.error("Finished Precondition retrying", e);
```

Comment:
This message seems a little confusing. It's an error, but the text sounds more like a success...  maybe

```java
log.error("Failed to meet preconditions. Existing...", e);
```